### ISSUE TITLE
Unable to Reproduce Reported Results

### ISSUE THREAD

**Issue by ThoCao** [Cedarch/M3Net#10]
Hi, thank you very much for releasing this excellent work and the full source code.
I really appreciate the clarity of the documentation and the pretrained model.

I was able to successfully run the evaluation script and download the complete NuScenes v1.0 dataset, and the environment appears to run without errors.

However, my reproduced results are far from the reported results in the paper (mAP = 0.0, NDS = 0.0), so I believe something may be wrong in my setup or there may be an issue in the released code. I would be grateful if you could help verify whether I missed something.

**Environment & What I Did**

**1. Dataset**
- Downloaded full NuScenes [v1.0 trainval](https://www.nuscenes.org/)
- Verified dataset structure (samples, sweeps, maps, v1.0-trainval)
- No missing files

**2. Pretrained model**
[ckpt/m3net_transformer.pth
](https://drive.google.com/file/d/1l9tU2YSWoNHWWc-MhJRJ1ssV3LdpK58j/view)

**3. Eval command (same as README)**

`# 8 gpus
python -m torch.distributed.launch --nproc_per_node=8 test.py --launcher pytorch \
--cfg_file cfgs/nuscenes_models/m3net_det_map_occ.yaml \
--batch_size 8 --workers 2 \
--ckpt /path/to/your/ckpt`

**❗ Issue: Evaluation Produces All-Zero Metrics**

`mAP: 0.0000
NDS: 0.0000

mATE: 1.0000
mASE: 1.0000
mAOE: 1.0000
mAVE: 1.0000
mAAE: 1.0000
`

**📄 Full Logs**

`
2025-11-16 07:13:21,241   INFO  ==> Loading parameters from checkpoint ckpt/m3net_transformer.pth to GPU
2025-11-16 07:13:21,662   INFO  ==> Checkpoint trained from version: pcdet+0.6.0+425cc1c+py77d20fb
2025-11-16 07:13:21,702   INFO  ==> Done (loaded 1120/1120)

2025-11-16 07:13:22,779   INFO  *************** EPOCH 3 EVALUATION *****************

eval:   0%|0/6019 [00:00<?, ?it/s]
ShapelyDeprecationWarning (from nuscenes.map_expansion.map_api)

eval:  29%|1747/6019 [21:48<55:23, 1.29it/s, recall_0.3=(0, 4858) / 55423]
eval:  75%|4514/6019 [56:08<19:32, 1.28it/s, recall_0.3=(0, 13033) / 129905]
eval: 100%|6019/6019 [1:15:17<00:00, 1.33it/s, recall_0.3=(0, 15447) / 149480]

2025-11-16 08:28:40,051   INFO  *************** Performance of EPOCH 3 *****************
2025-11-16 08:28:40,075   INFO  Generate label finished (sec_per_example: 0.7505)

recall_roi_0.3: 0.000000
recall_rcnn_0.3: 0.103338
recall_roi_0.5: 0.000000
recall_rcnn_0.5: 0.086794
recall_roi_0.7: 0.000000
recall_rcnn_0.7: 0.049251

Average predicted number of objects (6019 samples): 200.000

==================================================
Loading NuScenes tables (v1.0-trainval) ...
- 23 category
- 8 attribute
- 4 visibility
- 64386 instance
- 12 sensor
- 10200 calibrated_sensor
- 2631083 ego_pose
- 850 scene
- 34149 sample
Done loading in 33.4s

Reverse indexing ... done in 5.2s
==================================================

2025-11-16 08:30:46,825   INFO  Predictions saved to:
    /workspace/m3net/output/nuscenes_models/m3net_det_map_occ/default/eval/epoch_3/val/default/final_result/data/results_nusc.json

Initializing nuScenes detection evaluation...
Loaded detections for 6019 samples.
Loaded ground truth for 6019 samples.

Filtering predictions:
- Original boxes: 1,203,800
- After distance filter: 33,468
- After lidar-points filter: 33,468
- After bike-rack filter: 33,446

Filtering ground truth:
- Original boxes: 187,528
- After distance filter: 134,565
- After lidar-points filter: 121,871
- After bike-rack filter: 121,861

Calculating metrics...

======================= METRICS =======================

mAP: 0.0000
mATE: 1.0000
mASE: 1.0000
mAOE: 1.0000
mAVE: 1.0000
mAAE: 1.0000
NDS: 0.0000

-------------- Per-class AP --------------
car                    AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
truck                  AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
bus                    AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
trailer                AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
construction_vehicle   AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
pedestrian             AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
motorcycle             AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
bicycle                AP=0.000   ATE=1.000  ASE=1.000  AOE=1.000  AVE=1.000  AAE=1.000
traffic_cone           AP=0.000   ATE=1.000  ASE=1.000  AOE=nan    AVE=nan    AAE=nan
barrier                AP=0.000   ATE=1.000  ASE=1.000  AOE=nan    AVE=nan    AAE=nan

---------------- Average ----------------
trans_err:   1.0000
scale_err:   1.0000
orient_err:  1.0000
vel_err:     1.0000
attr_err:    1.0000
mAP:         0.0000
NDS:         0.0000

2025-11-16 08:31:22,832   INFO  Results saved to:
    /workspace/m3net/output/nuscenes_models/m3net_det_map_occ/default/eval/epoch_3/val/default

2025-11-16 08:31:22,833   INFO  **************** Evaluation done ****************

`

**Comment by Dengfenglai1997** [comment]
您好，请问您的问题解决了吗？我现在也遇到了同样的问题，可以交流一下吗
